src/main/scala/controllers/WeatherDataAutomation.scala

```diff
@@ -0,0 +1,143 @@
+package controllers
+
+import akka.actor.ActorSystem
+import models.WeatherData
+
+import java.sql.PreparedStatement
+import scala.concurrent.duration.DurationInt
+import scala.concurrent.{ExecutionContextExecutor, Future}
+import scala.io.Source
+import scala.util.Using
+
+/**
+ * The WeatherDataAutomation object fetches weather data for a list of cities from the OpenWeatherMap API and saves it to a PostgreSQL database.
+ * The data fetching and saving process is scheduled to repeat every hour.
+ */
+object WeatherDataAutomation extends App {
+
+  // Create an implicit ActorSystem
+  implicit val system: ActorSystem = ActorSystem("weather-data-automation")
+
+  // Define the execution context, which is responsible for managing the execution of asynchronous tasks.
+  implicit val executionContext: ExecutionContextExecutor = system.dispatcher
+
+  /**
+   * Fetches weather data for the specified city from the OpenWeatherMap API and saves it to the database.
+   * @param cityName The name of the city.
+   * @return A Future representing the completion of the operation.
+   */
+  def fetchDataAndSaveToDatabase(cityName: String): Future[Unit] = {
+    // Fetch weather data for the specified city
+    val responseFuture: Future[String] = OpenWeatherMapApiClient.fetchData(cityName)
+
+    responseFuture.flatMap { json =>
+      // Process the fetched JSON data and save it to the database
+      val weatherDataFuture: Future[Unit] = Future {
+        // Decode the JSON data into a case class WeatherData
+        val weatherData: Option[WeatherData] = decodeWeatherData(json)
+
+        weatherData match {
+          case Some(data) =>
+            insertDataToPostgres(data)
+            println(s"Data inserted successfully for $cityName")
+          case None =>
+            println(s"Failed to parse JSON for $cityName")
+        }
+      }
+
+      weatherDataFuture.recover {
+        case exception =>
+          println(s"An error occurred while processing data for $cityName: ${exception.getMessage}")
+      }
+    }.recover {
+      case exception =>
+        println(s"An error occurred while fetching data for $cityName: ${exception.getMessage}")
+    }
+  }
+
+  // Schedule the execution to repeat every hour
+  system.scheduler.scheduleAtFixedRate(0.seconds, 1.hour) {
+    () =>
+      val cityNames: Seq[String] = Source.fromFile("/Users/marcintubielewicz/Documents/programming/WeatherDataAnalyserApp/src/main/resources/city_names.txt").getLines().toList
+      for (cityName <- cityNames) {
+        fetchDataAndSaveToDatabase(cityName)
+      }
+  }
+
+  /**
+   * Decodes the JSON data into a case class WeatherData.
+   * @param json The JSON data as a string.
+   * @return An Option containing the decoded WeatherData if successful, or None if unsuccessful.
+   */
+  private def decodeWeatherData(json: String): Option[WeatherData] = {
+    import io.circe.generic.auto._
```

Comment:
why do you import it here, not at the beginning of the file?

Reply:
shifted to the top of the file